File: 01-data-model/Python/01_data_type.py
# Computational Biomedicine
# Module 01 — Data Model
# 01 — Data Types

# ---------------------------------------------------------
# 1. List containing different data types
# ---------------------------------------------------------

variant_list = [
    "chr1",
    10180,
    "rs201694901",
    "T",
    "C"
]

print(variant_list)
print(type(variant_list))

for value in variant_list:
    print(value, type(value))

File: 01-data-model/R/01_data_type.R
# Computational Biomedicine
# Module 01 — Data Model
# 01 — Data Types
#
# Objective:
# Explore basic data types and type coercion in R using a genomic variant.

# ---------------------------------------------------------
# 1. Atomic vector and type coercion
# ---------------------------------------------------------

variant_vector <- c(
  "chr1",
  10180,
  "rs201694901",
  "T",
  "C"
)

variant_vector
typeof(variant_vector)

# R atomic vectors are homogeneous.
# Because character values are present, the numeric value is
# coerced to character.

# ---------------------------------------------------------
# 2. Explicit integer
# ---------------------------------------------------------

position_integer <- 10180L

typeof(position_integer)

# R represents numeric literals as double by default.
# The suffix L explicitly creates an integer.

# ---------------------------------------------------------
# 3. Basic data types
# ---------------------------------------------------------

chromosome <- "chr1"
position <- 10180
is_variant <- TRUE
missing_value <- NA

typeof(chromosome)
typeof(position)
typeof(is_variant)
typeof(missing_value)